### ISSUE TITLE
[Feature] exclude option for check_cols snapshot

### ISSUE THREAD

**Issue by graciegoheen** [dbt-labs/dbt-core#10438]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Right now snapshots have `check_cols` - we could add a new config for the inverse, something like `check_exclude_cols`.

This would be similar to the incremental model config - `merge_exclude_columns`

**Comment by will-sylvera** [comment]
hey, if this feature is still relevant i'd be keen to have a go at it this weekend. not contributed here yet but have a bit of time and should be able to get something together

**Comment by graciegoheen** [comment]
That would be great @will-sylvera! Definitely still relevant :)

**Comment by willmaclean** [comment]
working on this now -- doing it from my personal GH account not my work.

**Comment by willmaclean** [comment]
@graciegoheen hey sorry, slightly dumb q here: whats your process for raising PRs across multiple repos? I have to make changes here and in the adapters repo [here](https://github.com/dbt-labs/dbt-adapters/blob/60005a0a2bd33b61cb65a591bc1604b1b3fd25d5/dbt/include/global_project/macros/materializations/snapshots/strategies.sql#L136). should I just raise PRs in both places at once?

**Comment by graciegoheen** [comment]
Yep! You can just raise both PRs at once, and we will review them together

**Comment by ataft** [comment]
@graciegoheen Are there any updates on this? I see two PR's that look like they're just sitting there ready to go. With the new dbt snapshots config YAML, I would think that this feature is even more critical because there's less control with no model.

**Comment by dbeatty10** [comment]
Pull requests:
- https://github.com/dbt-labs/dbt-core/pull/10488
- https://github.com/dbt-labs/dbt-adapters/pull/274

**Comment by matejnovak29** [comment]
Any updates on this one? I would love this feature!
